idem_provider_azurerm/states/azurerm/postgresql/server_security_alert_policy.py

```diff
@@ -0,0 +1,266 @@
+
+    :param server_name: The name of the server.
+
+    :param resource_group: The name of the resource group. The name is case insensitive.
+
+    :param policy_state: Specifies the state of the policy, whether it is enabled or disabled. Possible values include:
+        'Enabled', 'Disabled'.
+
+    :param disabled_alerts: Specifies an array of alerts that are disabled. Possible values are: 'Sql_Injection',
+        'Sql_Injection_Vulnerability', and 'Access_Anomaly'. It is import to note that the default value of this
+        parameter is [''].
+
+    :param email_addresses: Specifies an array of e-mail addresses to which the alert is sent. It is important to note
+        that the default value of this parameter is [''].
+
+    :param email_account_admins: A boolean value that specifies whether the alert is sent to the account
+        administrators or not.
+
+    :param storage_endpoint: Specifies the blob storage endpoint (e.g. https://MyAccount.blob.core.windows.net).
+        This blob storage will hold all Threat Detection audit logs.
+
+    :param storage_account_access_key: Specifies the identifier key of the Threat Detection audit storage account.
+
+    :param retention_days: Specifies the number of days to keep in the Threat Detection audit logs.
+
+    :param force_access_key: A Boolean flag that represents whether or not the storage account access ket value should
+        be updated. If it is set to True, then the password will be updated if the server already exists. If it is set
+        to False, then the password will not be updated unless other parameters also need to be updated.
+        Defaults to False.
+
+    :param connection_auth: A dict with subscription and authentication parameters to be used in connecting to the
+        Azure Resource Manager API.
+
+    Example usage:
+
+    .. code-block:: yaml
+
+        Ensure server security alert policy exists:
+            azurerm.postgresql.server_security_alert_policy.present:
+                - server_name: my_server
+                - resource_group: my_rg
+                - policy_state: 'Enabled'
+                - connection_auth: {{ profile }}
+
+    '''
+    name = 'Default'
+
+    ret = {
+        'name': name,
+        'result': False,
+        'comment': '',
+        'changes': {}
+    }
+
+    if not isinstance(connection_auth, dict):
+        ret['comment'] = 'Connection information must be specified via connection_auth dictionary!'
+        return ret
+
+    policy = await hub.exec.azurerm.postgresql.server_security_alert_policy.get(
+        server_name=server_name,
+        resource_group=resource_group,
+        azurearm_log_level='info',
+        **connection_auth
+    )
+
+    if 'error' not in policy:
+        if policy_state != policy.get('state'):
+            ret['changes']['state'] = {
+                'old': policy.get('state'),
+                'new': policy_state
+            }
+
+        if disabled_alerts is not None:
```

Comment:
`is not None` is only necessary if evaluating a boolean. if you expect a string/list/dict/etc. but default to `None` in the parameters then just use `if variable`